packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1631,6 +1631,23 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
 
     assert(_debugCheckLocalizations(appLocale));
 
+    final Localizations localizations = Localizations(
```

Comment:
```suggestion
    final Widget child = Localizations(
```

And then below if `data == null` just do something like
```dart
     child = MediaQueryFromWindow(
        child: localizations,
      );
```

Reply:
Makes sense 👍